src/compat/api/validation/encoding/programmable/pipeline_bind_group_compat.spec.ts

```diff
@@ -0,0 +1,373 @@
+      });
+      encoder.setBindGroup(0, bindGroup0);
+      encoder.setBindGroup(1, bindGroup1);
+      return { shouldSucceed: true, use: false };
+    },
+  },
+};
+const kBindCaseNames = keysOf(kBindCases);
+
+const kDrawUseCases: {
+  [key: string]: (t: ValidationTest, encoder: GPURenderCommandsMixin) => void;
+} = {
+  draw: (t: ValidationTest, encoder: GPURenderCommandsMixin) => {
+    encoder.draw(3);
+  },
+  drawIndexed: (t: ValidationTest, encoder: GPURenderCommandsMixin) => {
+    const indexBuffer = t.makeBufferWithContents(new Uint16Array([0, 1, 2]), GPUBufferUsage.INDEX);
+    encoder.setIndexBuffer(indexBuffer, 'uint16');
+    encoder.drawIndexed(3);
+  },
+  drawIndirect(t: ValidationTest, encoder: GPURenderCommandsMixin) {
+    const indirectBuffer = t.makeBufferWithContents(
+      new Uint32Array([3, 1, 0, 0]),
+      GPUBufferUsage.INDIRECT
+    );
+    encoder.drawIndirect(indirectBuffer, 0);
+  },
+  drawIndexedIndirect(t: ValidationTest, encoder: GPURenderCommandsMixin) {
+    const indexBuffer = t.makeBufferWithContents(new Uint16Array([0, 1, 2]), GPUBufferUsage.INDEX);
+    encoder.setIndexBuffer(indexBuffer, 'uint16');
+    const indirectBuffer = t.makeBufferWithContents(
+      new Uint32Array([3, 1, 0, 0, 0]),
+      GPUBufferUsage.INDIRECT
+    );
+    encoder.drawIndexedIndirect(indirectBuffer, 0);
+  },
+};
+const kDrawCaseNames = keysOf(kDrawUseCases);
+
+const kDispatchUseCases: {
+  [key: string]: (t: ValidationTest, encoder: GPUComputePassEncoder) => void;
+} = {
+  dispatchWorkgroups(t: ValidationTest, encoder: GPUComputePassEncoder) {
+    encoder.dispatchWorkgroups(1);
+  },
+  dispatchWorkgroupsIndirect(t: ValidationTest, encoder: GPUComputePassEncoder) {
+    const indirectBuffer = t.makeBufferWithContents(
+      new Uint32Array([1, 1, 1]),
+      GPUBufferUsage.INDIRECT
+    );
+    encoder.dispatchWorkgroupsIndirect(indirectBuffer, 0);
+  },
+};
+const kDispatchCaseNames = keysOf(kDispatchUseCases);
+
+export const g = makeTestGroup(ValidationTest);
+
+g.test('twoDifferentTextureViews,render_pass')
+  .desc(
+    `
+Tests that you can not use 2 different views of the same texture in a render pass in compat mode..
+
+- Test you can not use incompatible views in the same bindGroup
+- Test you can not use incompatible views in different bindGroups
+- Test you can bind the same view in different bindGroups
+- Test binding incompatible bindGroups is ok as long as they are fixed before draw/dispatch
+- Test binding incompatible bindGroups is ok if there's no draw/dispatch
+
+  The last 2 tests are to check validation happens at the correct time (draw/dispatch) and not
+  at finish or setBindGroup.
+    `
+  )
+  .params(u =>
+    u
+      .combine('encoderType', kRenderEncodeTypes)
+      .combine('bindCase', kBindCaseNames)
+      .combine('useCase', kDrawCaseNames)
+      .combine('textureType', kTextureTypes)
+      .filter(
+        // storage textures can't have 2 bind groups point to the same
```

Comment:
Should these actually be a test that expect failure instead of being skipped then? (Especially the compat-specific behavior.)

Reply:
There is no compat specific behavior for this case. You can't bind a storage texture to two or more bind points. You'll get an "aliasing disallowed" validation error. That error is already tested. The test that checks that you can't alias storage texture bindings will also be run in compat mode.

Reply:
Ack.